test/utils/shippable/shippable.sh

```diff
@@ -9,6 +9,28 @@ script="${args[0]}"
 
 test="$1"
 
+function get_default_image_version {
+    # Find the completion file and parse it to get the default test image version
+    # Returns 'latest' if unable to parse the completion file
+
+    version='latest'
+    completion_file="$(find test -type f -name 'docker.txt' | tail -n 1)"
+    if [[ -f "$completion_file" ]]; then
+        found_version=$(grep 'default' "$completion_file" | cut -d ':' -f 2 | cut -d ' ' -f 1)
+
+        if [[ -n "$found_version" ]]; then
+            version="$found_version"
+        fi
+    fi
+
+    echo "$version"
+}
+
+# Pull correct version of default test image only for certain tests
+if [[ "${script}" =~ (units|sanity|windows|network) ]]; then
+    docker pull "quay.io/ansible/default-test-container:$(get_default_image_version)"
+fi
+
```

Comment:
I would prefer we rely on `ansible-test` for the parsing of the `docker.txt` file and for the download of the image. This way we don't duplicate the existing logic in shell scripting.

There is already a `--docker-no-pull` argument, maybe we can just add something like: `--docker-pull-only`. Or even something more generical like `--prepare-env-only` that would also install the dependencies?